packages/clarity-js/src/data/metadata.ts

```diff
@@ -19,19 +19,29 @@ export function start(): void {
   data = {
     projectId: config.projectId || hash(location.host),
     userId: user(),
-    sessionId: s.id,
+    sessionId: s.session,
     pageNum: s.count
   }
 
+  // For backward compatibility; remove in future iterations (v0.6.11)
+  if (config.upload && typeof config.upload === Constant.String && (config.upload as string).indexOf(Constant.HTTPS) === 0) {
+    let url = config.upload as string;
+    config.server = url.substr(0, url.indexOf("/", Constant.HTTPS.length));
+    config.upload = config.server.length > 0 && config.server.length < url.length ? url.substr(config.server.length + 1) : url;
```

Comment:
Why do you truncate the URL to the server name only in this?

Reply:
Now that we use cookie to store session state - it seemed wasteful to cache the complete URL. Especially since cookies are sent along with every network requests. This is backward compatibility code. And in future, we can simplify by sending server variable once and then report or upload url can be relative paths.